bin/deepstate/core/fuzz.py

```diff
@@ -515,14 +576,21 @@ def _run(self, command: List[str]) -> int:
     # any OS-specific errors encountered
     except OSError as e:
       self._kill()
-      raise FuzzFrontendError(f"{self.fuzzer} run interrupted due to exception {e}.")
+      raise FuzzFrontendError(f"{self.name} run interrupted due to exception {e}.")
 
     # SIGINT stops fuzzer, but continues execution
     except KeyboardInterrupt:
       print(f"Killing fuzzer {self.name} with PID {self.proc.pid}")
       self._kill()
       return 1
 
+    except FuzzFrontendError as e:
+      raise e
```

Comment:
This is tricky and may be bug prone.

Consider those two things:
- `class FuzzFrontendError(AnalysisBackendError):`
- in `main` we do `except AnalysisBackendError as e:`

As a result, if the `try: ...` above results in `raise AnotherAnalysisBackendError` it will be catched only by `except Exception` and not forwarded upper to `main` like it wasn't expected?

Not sure here but maybe it would  be better to except on `AnalysisBackendError`.

Reply:
Good point